src/Configuration.php

```diff
@@ -1198,10 +1198,14 @@ public function setPager($pager)
      * If no Pager has been explicitly provided, and Pcntl is available, this
      * will default to `cli.pager` ini value, falling back to `which less`.
      *
-     * @return string|OutputPager
+     * @return string|OutputPager|null
      */
     public function getPager()
     {
+        if (\getenv('TERM') === 'dumb') {
+            return null;
+        }
+
         if (!isset($this->pager) && $this->usePcntl()) {
```

Comment:
Thanks for this!

I was thinking it might make more sense to put the check for a dumb terminal here, i.e. if a pager isn't explicitly defined, don't define one. That way we leave room for users who intentionally add a pager via config, but don't force a pager when it doesn't make sense.

Reply:
@bobthecow Thank you for your review. I made it return `false` instead of `null` that same as  860c4819a09a43712eaa88fee2348e6d5d094ffe .